ee/agent/storage/sqlite/keyvalue_store_sqlite.go

```diff
@@ -0,0 +1,292 @@
+package agentsqlite
+
+import (
+	"context"
+	"database/sql"
+	"embed"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang-migrate/migrate/v4"
+	sqlitemigrationdriver "github.com/golang-migrate/migrate/v4/database/sqlite"
+	"github.com/golang-migrate/migrate/v4/source/iofs"
+	_ "modernc.org/sqlite"
+)
+
+const (
+	// Supported tables -- all tables should have the same `name` and `value` columns
+	TableStartupSettings = "startup_settings"
+)
+
+var supportedTables = map[string]struct{}{
+	TableStartupSettings: {},
+}
+
+//go:embed migrations/*.sqlite
+var migrations embed.FS
+
+type sqliteStore struct {
+	conn          *sql.DB
+	rootDirectory string
+	tableName     string
+}
+
+// OpenRO opens a connection to the database in the given root directory; it does
+// not perform database creation or migration.
+func OpenRO(ctx context.Context, rootDirectory string, tableName string) (*sqliteStore, error) {
+	if _, ok := supportedTables[tableName]; !ok {
+		return nil, fmt.Errorf("unsupported table %s", tableName)
+	}
+
+	conn, err := sql.Open("sqlite", dbLocation(rootDirectory))
+	if err != nil {
+		return nil, fmt.Errorf("opening startup db in %s: %w", rootDirectory, err)
+	}
```

Comment:
You'd asked for the RO option to be more lightweight and to skip the validation when we'd discussed on slack, since we need to use the RO option early on in launcher startup.

Reply:
I was thinking about skipping migration -- I'm less sure what should happen if the DB file isn't present. Or doesn't match expected schema. Bailing early and getting nothing isn't wrong

Reply:
I guess that's probably the right way to look at it -- If the DB is wrong, we should bail out and get nothing. Maybe it doesn't exist yet, maybe it hasn't been udpated to the current schema. So whatever we need to catch that and move on

Reply:
👍 I will handle that in the next PR that starts to use the RO connection